io_scene_niftools/modules/nif_export/geometry/mesh/skin_partition.py

```diff
@@ -0,0 +1,574 @@
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# ***** END LICENSE BLOCK *****
+
+import logging
+import pyffi
+from pyffi.formats.nif import NifFormat
+
+
+def update_skin_partition(self,
+                        maxbonesperpartition=4, maxbonespervertex=4,
+                        verbose=0, stripify=True, stitchstrips=False,
+                        padbones=False,
+                        triangles=None, trianglepartmap=None,
+                        maximize_bone_sharing=False,
+                        part_sort_order=[]):
+    """Recalculate skin partition data.
+
+    :deprecated: Do not use the verbose argument.
+    :param maxbonesperpartition: Maximum number of bones in each partition.
+        The num_bones field will not exceed this number.
+    :param maxbonespervertex: Maximum number of bones per vertex.
+        The num_weights_per_vertex field will be exactly equal to this number.
+    :param verbose: Ignored, and deprecated. Set pyffi's log level instead.
+    :param stripify: If true, stripify the partitions, otherwise use triangles.
+    :param stitchstrips: If stripify is true, then set this to true to stitch
+        the strips.
+    :param padbones: Enforces the numbones field to be equal to
+        maxbonesperpartition. Also ensures that the bone indices are unique
+        and sorted, per vertex. Raises an exception if maxbonespervertex
+        is not equal to maxbonesperpartition (in that case bone indices cannot
+        be unique and sorted). This options is required for Freedom Force vs.
+        the 3rd Reich skin partitions.
+    :param triangles: The triangles of the partition (if not specified, then
+        this defaults to C{self.data.get_triangles()}.
+    :param trianglepartmap: Maps each triangle to a partition index. Faces with
+        different indices will never appear in the same partition. If the skin
+        instance is a BSDismemberSkinInstance, then these indices are used as
+        body part types, and the partitions in the BSDismemberSkinInstance are
+        updated accordingly. Note that the faces are counted relative to
+        L{triangles}.
+    :param maximize_bone_sharing: Maximize bone sharing between partitions.
+        This option is useful for Fallout 3.
+    :param part_sort_order: List of body part numbers in the order in which
+        they should appear, e.g. [5, 3, 6]. The first entry is counted. When
+        maximize_bone_sharing is true, sorts the parts within the shared bones,
+        and sorts the shared bone lists based on its first body part.
+    """
+    logger = logging.getLogger("pyffi.nif.nitribasedgeom")
```

Comment:
Should we use the normal logger

Reply:
For now I just kept it without Niftools-specific modification, so that it could be quickly ported to pyffi if we start using the most recent version.